lib/pf/Switch.pm

```diff
@@ -3550,6 +3551,39 @@ sub getCiscoAvPairAttribute {
     return ;
 }
 
+=head2 generateACLFromTemplate
+
+Generate an ACL from a template
+
+=cut
+
+sub generateACLFromTemplate {
+    my ($self, $t, $allow, $proto, $src_host, $src_port, $dst_host, $dst_port) = @_;
+    my $acl = pf::mini_template->new($t)->process({
+        allow => $allow,
+        proto => $proto,
+        src_host => $src_host // "",
+        src_port => $src_port // "",
+        dst_host => $dst_host // "",
+        dst_port => $dst_port // "",
+    });
+
+    # remove double spaces to cleanup ACL
+    $acl =~ s/  / /g;
+    return $acl;
+}
+
+=head2 generateACL
+
+Generate an ACL using the default template
+
+=cut
+
+sub generateACL {
+    my ($self, $allow, $proto, $src_host, $src_port, $dst_host, $dst_port) = @_;
+    return $self->generateACLFromTemplate($DEFAULT_ACL_TEMPLATE, $allow, $proto, $src_host, $src_port, $dst_host, $dst_port);
```

Comment:
To further generalize to so non-templated switch modules can use a different template.
```perl
$self->generateACLFromTemplate($self->aclTemplate(), $args);

sub aclTemplate { $DEFAULT_ACL_TEMPLATE }
```